plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/semantics/completion/SQLQueryCompletionContext.java

```diff
@@ -342,8 +344,16 @@ private List<SQLQueryCompletionItem> prepareObjectComponentCompletions(
                         components = Stream.empty();
                     }
 
-                    return components.filter(a -> (componentType == null || componentType.isInstance(a)) && a.getName().toLowerCase().contains(componentNamePart))
-                        .map(SQLQueryCompletionItem::forDbObject)
+                    return components.filter(a -> (componentType == null || componentType.isInstance(a))
+                            && a.getName().toLowerCase().contains(componentNamePart))
+                        .map(x -> {
+                            //todo transform to the map strategy if add one more case
+                            if (x instanceof DBSEntity dbsEntity) {
+                                return SQLQueryCompletionItem.forRealTable(dbsEntity, true);
+                            } else {
+                                return SQLQueryCompletionItem.forDbObject(x);
+                            }
+                        })
```

Comment:
Let's move it to the caller, like it's done in `collectObjectRecursively` by adding `Function<T, SQLQueryCompletionItem> completionItemFabric` parameter.
```suggestion
                    return components.filter(a -> (componentType == null || componentType.isInstance(a))
                            && a.getName().toLowerCase().contains(componentNamePart))
                        .map(c -> completionItemFabric.apply((T) c))
```